File: Export MS-Access table to MS-Excel Worksheet/[C#][VB]-Export MS-Access table to MS-Excel Worksheet/C# and VB/AccessConnections_cs/AccessInformation.cs
using System.Collections.Generic;
using System.Linq;
using System.Data.OleDb;
using System.Data;
using System;

namespace AccessConnections_cs
{
    /// <summary>
    /// Methods to obtain table names and columns for tables in a ms-access database.
    /// These are not optimized, did them for demo purposes only. They can be optimized :-)
    /// </summary>
    public class AccessInformation
    {
        /// <summary>
        /// Key is table name, values are columns for table
        /// </summary>
        public Dictionary<string, List<string>> ColumnDict { get; set; }
        /// <summary>
        /// Get user table names from database
        /// </summary>
        /// <param name="DatabaseName">Existing database path and name</param>
        /// <returns></returns>
        public List<string> TableNames(string DatabaseName)
        {
            List<string> Names = new List<string>();

            using (OleDbConnection cn = new OleDbConnection { ConnectionString = DatabaseName.BuildConnectionString() })
            {

                cn.Open();

                Names = cn.GetOleDbSchemaTable(OleDbSchemaGuid.Tables, null)
                    .AsEnumerable()
                    .Where((col) => col.Field<string>("TABLE_TYPE") == "TABLE")
                    .Select((data) => data.Field<string>("TABLE_NAME"))
                    .ToList();

            }

            return Names;

        }
        public void TryDropTable(string DatabaseName, string TableName)
        {
            var Names = TableNames(DatabaseName);
            if (Names.Contains(TableName))
            {
                using (OleDbConnection cn = new OleDbConnection { ConnectionString = DatabaseName.BuildConnectionString() })
                {
                    using (OleDbCommand cmd = new OleDbCommand { Connection = cn, CommandText = $"DROP TABLE {TableName}" })
                    {
                        cn.Open();
                        try
                        {
                            cmd.ExecuteNonQuery();
                        }
                        catch (Exception ex)
                        {
                            //
                            // Discard exception for this sample but for a real app the exception should be delt with.
                            //
                        }
                    }
                }
            }
        }
        /// <summary>
        /// Get column names for a specific table
        /// </summary>
        /// <param name="DatabaseName"></param>
        /// <param name="TableName"></param>
        /// <returns></returns>
        public List<string> GetColumnNames(string DatabaseName, string TableName)
        {
            List<string> Names = new List<string>();
            using (OleDbConnection cn = new OleDbConnection { ConnectionString = DatabaseName.BuildConnectionString() })
            {
                string[] filterValues = { null, null, TableName, null };
                cn.Open();

                Names = cn.GetSchema("Columns", filterValues).AsEnumerable().Select((data) => $"[{data.Field<string>("COLUMN_NAME")}]").ToList();
            }

            return Names;

        }
        /// <summary>
        /// Get columns for each table in database
        /// </summary>
        /// <param name="DatabaseName"></param>
        public void GetTableColumnInformation(string DatabaseName)
        {
            ColumnDict = new Dictionary<string, List<string>>();

            var Names = TableNames(DatabaseName);
            foreach (string table in Names)
            {
                var ColNames = GetColumnNames(DatabaseName, table);

                ColumnDict.Add(table, ColNames);
            }
        }
    }
}

File: Export MS-Access table to MS-Excel Worksheet/[C#][VB]-Export MS-Access table to MS-Excel Worksheet/C# and VB/AccessConnections_vb/AccessInformation.vb
Imports System.Data.OleDb

''' <summary>
''' Methods to obtain table names and columns for tables in a ms-access database.
''' These are not optimized, did them for demo purposes only. They can be optimized :-)
''' </summary>
Public Class AccessInformation
    ''' <summary>
    ''' Key is table name, values are columns for table
    ''' </summary>
    ''' <returns></returns>
    Public Property ColumnDict As Dictionary(Of String, List(Of String))
    ''' <summary>
    ''' Get user table names from database
    ''' </summary>
    ''' <param name="DatabaseName">Existing database path and name</param>
    ''' <returns></returns>
    Public Function TableNames(ByVal DatabaseName As String) As List(Of String)
        Dim Names As New List(Of String)

        Using cn As New OleDbConnection With {.ConnectionString = DatabaseName.BuildConnectionString}

            cn.Open()

            Names = cn.GetOleDbSchemaTable(OleDbSchemaGuid.Tables, Nothing) _
                .AsEnumerable.Where(Function(col) col.Field(Of String)("TABLE_TYPE") = "TABLE") _
                .Select(Function(data) data.Field(Of String)("TABLE_NAME")).ToList

        End Using

        Return Names

    End Function
    Public Sub TryDropTable(ByVal DatabaseName As String, ByVal TableName As String)
        Dim Names = TableNames(DatabaseName)
        If Names.Contains(TableName) Then
            Using cn As New OleDbConnection With {.ConnectionString = DatabaseName.BuildConnectionString}
                Using cmd As New OleDbCommand With {.Connection = cn, .CommandText = $"DROP TABLE {TableName}"}
                    cn.Open()
                    Try
                        cmd.ExecuteNonQuery()
                    Catch ex As Exception
                        '
                        ' Discard exception for this sample but for a real app the exception should be delt with.
                        '
                    End Try
                End Using
            End Using
        End If
    End Sub
    ''' <summary>
    ''' Get column names for a specific table
    ''' </summary>
    ''' <param name="DatabaseName"></param>
    ''' <param name="TableName"></param>
    ''' <returns></returns>
    Public Function GetColumnNames(ByVal DatabaseName As String, ByVal TableName As String) As List(Of String)
        Dim Names As New List(Of String)
        Using cn As New OleDbConnection With {.ConnectionString = DatabaseName.BuildConnectionString}
            Dim filterValues As String() = {Nothing, Nothing, TableName, Nothing}
            cn.Open()

            Try
                Names = cn.GetSchema("Columns", filterValues).AsEnumerable.Select(Function(data) $"[{data.Field(Of String)("COLUMN_NAME")}]").ToList
            Catch ex As Exception
                Console.WriteLine(ex.Message)
            End Try

        End Using

        Return Names

    End Function
    ''' <summary>
    ''' Get columns for each table in database
    ''' </summary>
    ''' <param name="DatabaseName"></param>
    Public Sub GetTableColumnInformation(ByVal DatabaseName As String)
        ColumnDict = New Dictionary(Of String, List(Of String))

        Dim Names = TableNames(DatabaseName)
        For Each table As String In Names
            Dim ColNames = GetColumnNames(DatabaseName, table)

            ColumnDict.Add(table, ColNames)
        Next

    End Sub
End Class
